packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2466,6 +2491,382 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+      // just treated as a quoted ampersand.
+      if (label[i] != '&' || i == label.length - 1) {
+        displayLabel.write(label[i]);
+        continue;
+      }
+      final String acceleratorCharacter = label[i + 1];
+      if (acceleratorIndex == -1 && acceleratorCharacter != '&') {
+        acceleratorIndex = i - quotedAmpersands;
+      }
+      displayLabel.write(acceleratorCharacter);
+      // As we encounter '&&' pairs, the later indices must be adjusted so that
+      // they correspond with indices in the stripped string.
+      quotedAmpersands += 1;
+      i += 1; // Skip over the character after the ampersand.
+    }
+    setIndex?.call(acceleratorIndex);
+    return displayLabel.toString();
+  }
+
+  @override
+  State<MenuAcceleratorLabel> createState() => _MenuAcceleratorLabelState();
+
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.info}) {
+    return '$MenuAcceleratorLabel("$label")';
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('label', label));
+  }
+}
+
+class _MenuAcceleratorLabelState extends State<MenuAcceleratorLabel> {
+  late String _displayLabel;
+  int _acceleratorIndex = -1;
+  MenuAcceleratorCallbackBinding? _binding;
+  _MenuAnchorState? _anchor;
+  ShortcutRegistry? _shortcutRegistry;
+  ShortcutRegistryEntry? _shortcutRegistryEntry;
+  bool _showAccelerators = false;
+
+  @override
+  void initState() {
+    super.initState();
+    if (_platformSupportsAccelerators()) {
+      final Set<LogicalKeyboardKey> keysPressed = RawKeyboard.instance.keysPressed;
+      _showAccelerators = keysPressed.contains(LogicalKeyboardKey.altLeft) ||
+        keysPressed.contains(LogicalKeyboardKey.altRight);
+      RawKeyboard.instance.addListener(_handleKeyEvent);
+    }
+    _updateDisplayLabel();
+  }
+
+  @override
+  void dispose() {
+    assert(_platformSupportsAccelerators() || _shortcutRegistryEntry == null);
+    _displayLabel = '';
+    if (_platformSupportsAccelerators()) {
+      _shortcutRegistryEntry?.dispose();
+      _shortcutRegistryEntry = null;
+      _shortcutRegistry = null;
+      _anchor = null;
+      RawKeyboard.instance.removeListener(_handleKeyEvent);
+    }
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_platformSupportsAccelerators()) {
+      return;
+    }
+    _binding = MenuAcceleratorCallbackBinding.maybeOf(context);
+    _anchor = _MenuAnchorState._maybeOf(context);
+    _shortcutRegistry = ShortcutRegistry.maybeOf(context);
+    _updateAcceleratorShortcut();
+  }
+
+  @override
+  void didUpdateWidget(MenuAcceleratorLabel oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.label != oldWidget.label) {
+      _updateDisplayLabel();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Nit: I think this build method should be the last thing in this file [according to the styleguide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense) unless I'm mistaken.